script.js

```diff
@@ -1,4 +1,177 @@
+//Global variables
+var playerOne = 0;
+var playerTwo = 0;
+var player1Turn = true;
+var player2Turn = false;
+var diceOne = 0;
+var diceTwo = 0;
+var GAME_STATUS = "START_GAME";
+var gameMode = "";
+var winningPlayer = "";
+
+//This function generates random dice roll
+var rollDice = function () {
+  var diceNumber = Math.floor(Math.random() * 6) + 1;
+  return diceNumber;
+};
+
+//This function generate current score for current player
+var getScore = function (input) {
+  var score = 0;
+
+  console.log(diceOne);
+  console.log(diceTwo);
+
+  if (input == "1") {
+    score = diceOne.toString() + diceTwo.toString();
+  } else if (input == "2") {
+    score = diceTwo.toString() + diceOne.toString();
+  }
+  console.log(score);
+  return score;
+};
+
+//This function rolled 2 dice and ask user to enter option 1 or 2
+var getCurrentRoll = function () {
+  diceOne = rollDice();
+  diceTwo = rollDice();
+
+  var msg =
+    "You have rolled: " +
+    diceOne +
+    " and " +
+    diceTwo +
+    "<br> Please choose option 1 or 2.";
+
+  GAME_STATUS = "ASK_OPTION";
+  return msg;
+};
+
+var checkWinningPlayer = function (gameMode) {
+  if (gameMode == "HIGH") {
+    if (playerOne > playerTwo) {
+      winningPlayer = "Player 1";
+    }
+    if (playerTwo > playerOne) {
+      winningPlayer = "Player 2";
+    }
+  }
+
+  if (gameMode == "LOW") {
+    if (playerOne < playerTwo) {
+      winningPlayer = "Player 1";
+    }
+    if (playerTwo < playerOne) {
+      winningPlayer = "Player 2";
+    }
+  }
+  return winningPlayer;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var currentScore = 0;
+
+  //Ask user to chose game mode between highest combined number or lowest combine number
+  if (GAME_STATUS == "START_GAME") {
+    var startMsg =
+      "Let's start by choosing which game mode you want to play. <br> Please input 'HIGH' for highest combined number of dice roll or 'LOW' for lowest combined number of dice roll.";
+    GAME_STATUS = "CHOOSE_MODE";
+    return startMsg;
+  }
+
+  //save user input for the game mode only if game status is at the start of the game
+  if (GAME_STATUS == "CHOOSE_MODE") {
+    console.log(input);
+    if (input.toUpperCase() == "HIGH") {
+      gameMode = "HIGH";
+      console.log(gameMode);
+    }
+    if (input.toUpperCase() == "LOW") {
+      gameMode = "LOW";
+      console.log(gameMode);
+    } else if (input.toUpperCase() != "HIGH" && input.toUpperCase != "LOW") {
+      var invalid =
+        "You have entered invalid choice for game mode. Please input either HIGH or LOW. Click Roll Dice to re-start the game";
+      GAME_STATUS = "START_GAME";
+      return invalid;
+    }
+    console.log(gameMode);
```

Comment:
Good use of console logs as a way to check certain values at different points of the game. Do consider pairing/concatanating them with strings so you can trace these in the console